wgpu-hal/src/metal/device.rs

```diff
@@ -1611,3 +1658,12 @@ impl crate::Device for super::Device {
         Ok(())
     }
 }
+
+// TODO: `newComputePipelineStateWithDescriptor:error:` is not exposed on
+// `MTLDevice`, is this always correct?
+fn new_compute_pipeline_state_with_descriptor(
+    device: &ProtocolObject<dyn MTLDevice>,
+    descriptor: &MTLComputePipelineDescriptor,
+) -> Result<Retained<ProtocolObject<dyn MTLComputePipelineState>>, Retained<NSError>> {
+    unsafe { msg_send![device, newComputePipelineStateWithDescriptor: descriptor, error: _] }
+}
```

Comment:
nitpick: I'd prefer that this be inlined to its single use site; the comment is welcome, and we should validate it (CC @gfx-rs/wgpu), but the function signature simply doesn't seem to carry its weight.

Reply:
I've inlined it now.